src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -178,6 +178,8 @@ def resolve(self, root_reqs, check_supported_wheels):
                         gone_in="21.1",
                         issue=8711,
                     )
+
+                # is a local sdist or path -- reinstall
```

Comment:
This case did cover the editable reinstall from local directory. This PR alsos covers editable reinstall of VCS urls :+1: